libbeat/beat/beat.go

```diff
@@ -218,7 +217,7 @@ func (b *Beat) Run() error {
 	// Callback is called if the processes is asked to stop.
 	// This needs to be called before the main loop is started so that
 	// it can register the signals that stop or query (on Windows) the loop.
-	service.HandleSignals(b.Exit)
+	service.HandleSignals(b.Stop)
```

Comment:
An alternative here could be:

```

var callback sync.Once

// Exiting beat -> shutdown
func (b *Beat) Exit() {

	callback.Do(func() {
		logp.Info("Start exiting beat")
		close(b.exit)
	})
}
```

This makes sure Exit is only called once. It will not solve the problem that the filebeat and topbeat Stop function to not cleanup everything yet before stopping.